### PR TITLE
feat: add geometry about relation plots

### PR DESCRIPTION
**Pull request by hustcc**
image

- [x] fix transform ci
- [x] add tree, cluster transform
- [x] add link geometry and link shape

image

- [x] add node geometry
   - [x] point shape
   - [x] polygon shape
- [x] add edge geometry
   - [x] edge with arrow
   - [x] hvh/vhv
   - [x] smooth

image
image
image
image
image

   - [x] arc for 2 points
     - [x] coordinate

image
image

   - [x] ribbon for 4 points (sankey, chord)
     - [x] coordinate

image
image

- [x] remove ribbon from polygon shape

### CODE REVIEW

File: src/shape/link/link.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { ShapeComponent as SC } from '../../runtime';
+import { Edge } from '../../shape/edge/edge';
+
+export type LinkOptions = Record<string, any>;
+
+// Draw a line from `start` to `end`.
+export const Link: SC<LinkOptions> = (...args) => {
+  return Edge(...args);
```
Comment (hustcc) [inline src/shape/link/link.ts:8]: 最好改成 Edge 引用 Link。

File: __tests__/unit/transform/connector.spec.ts [hunk 2]
```diff
@@ -10,9 +10,13 @@ describe('connector', () => {
...
-    expect(c1({ data })).toBe(data);
+    let r = await c1({ data });
+    expect(r.data).toBe(data);
 
-    const c2 = Connector({ callback: (d) => d.filter((i) => i.a === 1) });
-    expect(c2({ data })).toEqual([{ a: 1, b: 2, c: 3 }]);
+    const c2 = Connector({
+      callback: (d) => d.filter((i) => i.a === 1),
+    });
+    r = await c2({ data });
```
Comment (pearmini) [inline __tests__/unit/transform/connector.spec.ts:19]: 两个变量或者直接写在函数体里面

File: src/shape/link/link.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+};
```
Comment (pearmini) [inline src/shape/link/link.ts:9]: edge 引入 link

File: src/shape/node/polygon.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { ShapeComponent as SC } from '../../runtime';
+import { Polygon as PolygonShape } from '../polygon/polygon';
+
+export type PolygonOptions = Record<string, any>;
+
+export const Polygon: SC<PolygonOptions> = (...args) => {
+  return PolygonShape(...args);
+};
```
Comment (pearmini) [inline src/shape/node/polygon.ts:8]: 之后可以改成 interval 的 rect

**Review by pearmini — CHANGES_REQUESTED** [review]

**Pull request merged**